editor.planx.uk/src/pages/FlowEditor/components/Settings/Submissions/EventsLog.tsx

```diff
@@ -0,0 +1,170 @@
+import KeyboardArrowDown from "@mui/icons-material/KeyboardArrowDown";
+import KeyboardArrowUp from "@mui/icons-material/KeyboardArrowUp";
+import Payment from "@mui/icons-material/Payment";
+import Send from "@mui/icons-material/Send";
+import Avatar from "@mui/material/Avatar";
+import Box from "@mui/material/Box";
+import Collapse from "@mui/material/Collapse";
+import IconButton from "@mui/material/IconButton";
+import { styled } from "@mui/material/styles";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableHead from "@mui/material/TableHead";
+import TableRow from "@mui/material/TableRow";
+import Typography from "@mui/material/Typography";
+import DelayedLoadingIndicator from "components/DelayedLoadingIndicator";
+import ErrorFallback from "components/ErrorFallback";
+import { format } from "date-fns";
+import React, { useState } from "react";
+
+import { GetSubmissionsResponse, Submission } from ".";
+
+const ErrorSummary = styled(Box)(({ theme }) => ({
+  marginTop: theme.spacing(1),
+  padding: theme.spacing(3),
+  border: `5px solid ${theme.palette.error.main}`,
+}));
+
+const EventsLog: React.FC<GetSubmissionsResponse> = ({
+  submissions,
+  loading,
+  error,
+}) => {
+  if (loading)
+    return (
+      <DelayedLoadingIndicator
+        msDelayBeforeVisible={0}
+        text="Fetching events..."
+      />
+    );
+  if (error) return <ErrorFallback error={error} />;
+  if (submissions.length === 0)
+    return (
+      <ErrorSummary role="status" data-testid="error-summary">
+        <Typography variant="h4" gutterBottom>
+          {`No payments or submissions found for this service`}
+        </Typography>
+        <Typography variant="body2">
+          {`If you're looking for events before January 1, 2024, please contact a PlanX developer.`}
+        </Typography>
+      </ErrorSummary>
+    );
+
+  return (
+    <TableContainer>
+      <Table>
+        <TableHead>
+          <TableRow>
+            <TableCell sx={{ width: 300 }}>
+              <strong>Event</strong>
+            </TableCell>
+            <TableCell sx={{ width: 200 }}>
+              <strong>Date</strong>
+            </TableCell>
+            <TableCell>
+              <strong>Session ID</strong>
+            </TableCell>
+            <TableCell></TableCell>
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {submissions.map((submission) => (
+            <CollapsibleRow {...submission} />
```

Comment:
There's a few ways to make each even instance of this shaded I think.

If native CSS in a styled component won't work with something like `&:nth-of-type(even)` in a styled component (I would think is possible), then we could achieve this with JS.

```tsx
<TableBody>
  {submissions.map((submission, i ) => (
    <CollapsibleRow {...submission} isEven={(i + 1) % 2} />
  ))}
</TableBody>
```

And then we can use isEven is our `sx` prop.

Reply:
This would be achievable in CSS using `:nth-of-type(4n+3)`, which is applying to every 4th row (every other summary row) starting at row three (the second summary row).

However as we're using a shaded background for the expanding element, I think we'd be best to leave the summary rows as white background.